check/lib/Util.hs

```diff
@@ -0,0 +1,136 @@
+module Util (inCabalNewRepl, normalizeLines, normalizeSignatures) where
+
+import Data.Char (isAlphaNum, isDigit, isUpper)
+import System.Environment (getEnvironment)
+import System.Process
+import qualified Text.Parsec as P
+import qualified Text.Parsec.String as P
+
+inCabalNewRepl :: String -> String -> IO String
+inCabalNewRepl module_ input_ = do
+    -- if HC variable is set, we use it to invoke 'cabal new-repl'
+    envvars <- getEnvironment
+    let mhc = lookup "HC" envvars
+    let withCompiler = maybe [] (\hc -> ["-w", hc]) mhc
+
+    putStrLn "=== INPUT ==="
+    putStrLn input
+    output <- readProcess "cabal" (args withCompiler) input
+    putStrLn "=== OUTPUT ==="
+    putStrLn output
+
+    return output
+  where
+    args withCompiler =
+        [ "new-repl"
+        , "for-repl"
+        , "--ghc-option=-ignore-dot-ghci"
+        , "--enable-tests"
+        , "--enable-benchmarks"
+        ] ++ withCompiler ++
+        [ "--"
+        ]
+
+    input = unlines
+        -- load module
+        [ ":set prompt \"\""
+        , ":l " ++ module_
+        , show randomString1
+        ]
+        ++ input_ ++
+        unlines
+            [ "", show randomString2
+            ]
+
+-- markers
+randomString1 :: String
```

Comment:
Please document what these random strings are for, since it's not obvious to me.

Reply:
done.